src/radical/entk/appman/appmanager.py

```diff
@@ -958,3 +926,34 @@ def pipeline_update(msg, reply_to, corr_id, mq_channel):
             raise
 
     # ------------------------------------------------------------------------------------------------------------------
+
+    def _terminate(self):
```

Comment:
I don't see a reason why we should have a separate _terminate method. I would be in favor of putting the contents of the current _terminate() under terminate() and removing _terminate().

Reply:
I did it as we said. Check comment in issue #330

Reply:
Okay, let's just the keep public method. Move the contents of the private to the public method. resource_terminate() can call terminate(). Do you want me to do it or do you want to take it?